src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int MAX_CAPACITY = 10;
+    private static final float GROWTH_COEFFICIENT = 1.5f;
+    private static final String NO_ELEMENT = "No such element in the list";
+    private static final String INDEX_ERROR = "The index is out of list`s interval";
+    private Object[] data;
+    private int size;
 
+    public ArrayList() {
+        data = new Object[MAX_CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        if (size == data.length) {
+            increaseCapacity();
+        }
+        data[size++] = value;
+    }
 
+    @Override
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        if (size == data.length) {
+            increaseCapacity();
+        }
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_ERROR);
+        }
+        System.arraycopy(data, index, data, index + 1, size - index);
+        data[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public T get(int index) throws ArrayListIndexOutOfBoundsException {
+        checkIfIndexInInterval(index);
+        return (T) data[index];
     }
 
     @Override
-    public void set(T value, int index) {
-
+    public T remove(int index) throws ArrayListIndexOutOfBoundsException {
+        checkIfIndexInInterval(index);
+        // mvn попросив зробити змінну final, чи так правильно?
+        final T elementToRemove = (T) data[index];
+        if (size == data.length) {
```

Comment:
```suggestion
        if ...
```
this block don't need this
is to delete an element